core/src/test/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectLifecycleSpec.scala

```diff
@@ -132,4 +165,126 @@ class GoogleBillingProjectLifecycleSpec extends AnyFlatSpec {
         ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
       )
   }
+
+  it should "store the billing profile ID during billing project creation" in {
+    val repo = mock[BillingRepository]
+    val bpm = mock[BillingProfileManagerDAO]
+    val samDAO = mock[SamDAO]
+    val wsmResourceRecordDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    val bp = new GoogleBillingProjectLifecycle(repo, bpm, samDAO, mock[GoogleServicesDAO])
+
+    val createRequest = CreateRawlsV2BillingProjectFullRequest(
+      RawlsBillingProjectName("fake_project_name"),
+      Some(RawlsBillingAccountName("fake_billing_account_name")),
+      None,
+      None,
+      None,
+      None
+    )
+
+    when(
+      bpm.createBillingProfile(
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(createRequest.billingInfo),
+        any(),
+        ArgumentMatchers.eq(testContext)
+      )
+    )
+      .thenReturn(profileModel)
+
+    when(
+      samDAO.syncPolicyToGoogle(
+        ArgumentMatchers.eq(SamResourceTypeNames.billingProject),
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
+      )
```

Comment:
I think that if all of the argument matchers are `eq` you can simplify as follows:
```suggestion
      samDAO.syncPolicyToGoogle(
        SamResourceTypeNames.billingProject,
        createRequest.projectName.value,
        SamBillingProjectPolicyNames.owner
      )
```